File: octopoints_flutter/lib/db/entity/rule_entity.dart
import 'package:floor/floor.dart';
import 'package:octopoints_flutter/db/database.dart';

@Entity(tableName: 'rules', foreignKeys: [
  ForeignKey(
    childColumns: ['matchId'],
    parentColumns: ['id'],
    entity: MatchEntity,
    onDelete: ForeignKeyAction.cascade,
  )
], indices: [Index(value: ["matchId"], unique: true)])
class RuleEntity {
  @PrimaryKey(autoGenerate: true)
  final int? id;
  final int matchId;
  final int winners;
  final int total;

  RuleEntity(
    this.id,
    this.matchId,
    this.winners,
    this.total,
  );
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/entity/RuleEntity.kt
package dev.tpavani.octopoints.db.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "rules",
    foreignKeys = [
        ForeignKey(
            entity = MatchEntity::class,
            parentColumns = ["id"],
            childColumns = ["matchId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [
        Index(
            value = ["matchId"],
            unique = true
        ),
    ],
)
class RuleEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Int,
    val matchId: Int,
    var winners: Int,
    var total: Int,
    var battleRoyale: Boolean,
)